fastcrypto-zkp/src/conversions.rs

```diff
@@ -154,12 +164,25 @@ pub fn blst_g1_affine_to_bls_g1_affine(pt: &blst_p1_affine) -> BlsG1Affine {
     unsafe {
         blst_p1_affine_serialize(out.as_mut_ptr(), pt);
     }
-    let infinity = out[0] & (1 << 6) != 0;
-    BlsG1Affine::new(
-        Fp384::from_be_bytes_mod_order(&out[..48]),
-        Fp384::from_be_bytes_mod_order(&out[48..]),
-        infinity,
-    )
+    let infinity = out[0] == 0x40;
+    println!("out: {:?}, infinity: {:?}", out, infinity);
+    // BLST disagrees with Arkworks on the uncompressed representation of the infinity point on G1
+    // https://github.com/supranational/blst/blob/6382d67c72119d563975892ed49ba32e92d3d0da/src/e1.c#L153-L162
+    // the infinity bit notwithstanding,
+    // Arkworks: x = Fq::zero(), y = Fq::one()
+    // BLST: x = y = Fq::zero()
+    // BLST follows the standard here (see https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-pairing-friendly-curves-10#appendix-C)
+    let y = if infinity {
+        Fq::one()
+    } else {
+        Fp384::from_be_bytes_mod_order(&out[48..])
+    };
+    let x = {
+        // Mask away the flag bits (which Arkworks doesn't use)
+        out[0] &= 0b0001_1111;
```

Comment:
I read the RFC, thanks, it makes sense. Because the curve uses 381 bits, the first 3 bits in a 384bit stream (48bytes) are used as flags, but Arkworks didn't take that into account and instead they used the 0,1 point as infinity :(